File: Solution/UISolution/tflAssessmentReact/src/InterviewDetails/services/InterviewService.js
 const API_URL = "http://localhost:5129/api/interviews";

 // Centralized response handler
 const handleResponse = async (response) => {
   if (!response.ok) {
     const errorDetails = await response.json();
     throw new Error(errorDetails.message || "Something went wrong");
   }
   return await response.json();
 };
 
 class InterviewService {
   constructor() {
     this.apiBaseUrl = API_URL;
   }
 
   // Fetch all interview candidates
   async getAllInterviewCandidates() {
     try {
       const response = await fetch(`${this.apiBaseUrl}/details`);
       return await handleResponse(response);
     } catch (error) {
       console.error('Error fetching interview candidates:', error);
       throw error;
     }
   }
 
   // Fetch interviewed candidates by candidateId
   async getInterviewedCandidatesSubjects(candidateId) {
     try {
       const response = await fetch(`${this.apiBaseUrl}/candidate/${candidateId}`);
       return await handleResponse(response);
     } catch (error) {
       console.error(`Error fetching candidate subjects for ${candidateId}:`, error);
       throw error;
     }
   }
 
   // Fetch interview details by interviewId
   async getInterviewDetails(interviewId) {
     try {
       const response = await fetch(`${this.apiBaseUrl}/${interviewId}`);
       return await handleResponse(response);
     } catch (error) {
       console.error(`Error fetching interview details for ${interviewId}:`, error);
       throw error;
     }
   }
 
   // Reschedule interview (PUT request)
   async rescheduleInterview(interviewId, time, date) {
     try {
       const response = await fetch(
         `${this.apiBaseUrl}/${interviewId}/time/${encodeURIComponent(time)}/date/${date.toISOString().split('T')[0]}`, 
         {
           method: 'PUT',
           headers: { 'Content-Type': 'application/json' }
         }
       );
       return await handleResponse(response);
     } catch (error) {
       console.error(`Error rescheduling interview ${interviewId}:`, error);
       throw error;
     }
   }
 
   // Change interviewer (PUT request)
   async changeInterviewer(interviewId, smeId) {
     try {
       const response = await fetch(
         `${this.apiBaseUrl}/${interviewId}/subjectexperts/${smeId}`, 
         {
           method: 'PUT',
           headers: { 'Content-Type': 'application/json' }
         }
       );
       return await handleResponse(response);
     } catch (error) {
       console.error(`Error changing interviewer for ${interviewId}:`, error);
       throw error;
     }
   }
 
   // Cancel interview (DELETE request)
   async cancelInterview(interviewId) {
     try {
       const response = await fetch(`${this.apiBaseUrl}/${interviewId}`, {
         method: 'DELETE',
         headers: { 'Content-Type': 'application/json' }
       });
       return await handleResponse(response);
     } catch (error) {
       console.error(`Error cancelling interview ${interviewId}:`, error);
       throw error;
     }
   }
 }
 
 export default new InterviewService();
 

File: Solution/WebSolution/AssessmentPortal/Monolithic/Monolithic_Controller/AssessmentServices/Implementations/InterviewService.cs
using Transflower.TFLAssessment.Entities;
using Transflower.TFLAssessment.Services.Interfaces;
using Transflower.TFLAssessment.Repositories.Interfaces;

namespace Transflower.TFLAssessment.Services;


public class InterviewService : IInterviewService
{
    private readonly IInterviewRepository _repository;
    public InterviewService(IInterviewRepository repository)
    {
        _repository = repository;
    }
    public async Task<List<InterviewCandidateDetails>> GetAllInterviewCandidates()
    {
        return await _repository.GetAllInterviewCandidates();
    }
    public async Task<List<InterviewCandidateDetails>> GetInterviewedCandidatesSubjects(int candidateId)
    {
        return await _repository.GetInterviewedCandidatesSubjects(candidateId);
    }
    public async Task<InterviewDetails> GetInterviewDetails(int interviewId)
    {
        return await _repository.GetInterviewDetails(interviewId);
    }
    public async Task<bool> RescheduleInterview(int interviewId, DateTime date)
    {
        return await _repository.RescheduleInterview(interviewId, date);
    }
    public async Task< bool> RescheduleInterview(int interviewId, string time)
    {
        return await _repository.RescheduleInterview(interviewId, time);
    }
    public async Task<bool> RescheduleInterview(int interviewId, string time, DateTime date)
    {
        return await _repository.RescheduleInterview(interviewId, time, date);
    }
    public async Task<bool> ChangeInterviewer(int interviewId, int smeId)
    {
        return await _repository.ChangeInterviewer(interviewId, smeId);
    }
    public async  Task<bool> CancelInterview(int interviewId)
    {
        return await _repository.CancelInterview(interviewId);
    }
}
